src/query/storage/m3/storage.go

```diff
@@ -97,11 +97,17 @@ func (s *m3storage) FetchProm(
 	}
 
 	defer accumulator.Close()
-	result, _, err := accumulator.FinalResultWithAttrs()
+	result, attrs, err := accumulator.FinalResultWithAttrs()
 	if err != nil {
 		return storage.PromResult{}, err
 	}
 
+	resolutions := make([]int64, 0, len(attrs))
```

Comment:
Does it make sense perhaps to make resolution `time.Duration` rather than `int64`? It would be more explicit (`time.Duration` is just `int64` under the hood anyhow too).